atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/Sweeper.java

```diff
@@ -15,30 +15,28 @@
  */
 package com.palantir.atlasdb.sweep;
 
-import java.util.Set;
 import java.util.function.LongSupplier;
 
 import com.google.common.base.Optional;
-import com.google.common.collect.ImmutableSet;
 import com.palantir.atlasdb.keyvalue.api.Value;
 import com.palantir.atlasdb.protos.generated.TableMetadataPersistence;
 
 public enum Sweeper {
     CONSERVATIVE((unreadableTs, immutableTs) -> Math.min(unreadableTs.getAsLong(), immutableTs.getAsLong()),
-                 ImmutableSet.of(Value.INVALID_VALUE_TIMESTAMP),
+                 new long[] {Value.INVALID_VALUE_TIMESTAMP},
                  false,
                  true),
     THOROUGH((unreadableTs, immutableTs) -> immutableTs.getAsLong(),
-             ImmutableSet.of(),
+             new long[] {},
              true,
              false);
 
     private final SweepTimestampSupplier sweepTimestampSupplier;
-    private final Set<Long> timestampsToIgnore;
+    private final long[] timestampsToIgnore;
```

Comment:
CandidateCellForSweeping uses an array of primitive longs to store timestamps for memory efficiency reasons. Every call to Set.contains() with a primitive long would auto-box the argument, resulting in an extra memory allocation per every timestamp you process.